src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,200 @@
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't delete a car by id. " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String query = "SELECT cd.driver_id, cd.car_id, c.model, c.manufacturer_id "
+                + "FROM cars c "
+                + "JOIN cars_drivers cd ON cd.car_id = c.id "
+                + "WHERE driver_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, driverId);
+            ResultSet resultSet = statement.executeQuery();
+            List<Car> cars = new ArrayList<>();
+            while (resultSet.next()) {
+                Car car = new Car();
+                car.setId(resultSet.getObject("car_id", Long.class));
+                car.setModel(resultSet.getString("model"));
+                cars.add(car);
+            }
+            return cars;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get all cars", e);
+        }
+    }
+
+    private void insertDriver(Car car) {
+        String query = "INSERT INTO cars_drivers (car_id, driver_id) VALUES (?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, car.getId());
+            for (Driver driver : car.getDrivers()) {
+                statement.setLong(2, driver.getId());
+                statement.executeUpdate();
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't insert a driver to a car. " + car, e);
+        }
+    }
+
+    private Car getCarWithManufacturer(ResultSet resultSet) throws SQLException {
+        Long manufacturerId = resultSet.getObject("manufacturer_id", Long.class);
+        String name = resultSet.getString("manufacturer");
+        String country = resultSet.getString("country");
+        Manufacturer manufacturer = new Manufacturer(manufacturerId, name, country);
+        Car car = new Car();
+        car.setId(resultSet.getObject("car_id", Long.class));
+        car.setModel(resultSet.getString("car_model"));
+        car.setManufacturer(manufacturer);
+        return car;
+    }
+
+    private List<Driver> getAllDriversForCar(Long carId) {
+        String query = "SELECT cd.car_id, cd.driver_id, d.name as driver_name, d.license_number "
+                + "FROM drivers d "
+                + "JOIN cars_drivers cd ON cd.driver_id = d.id "
+                + "WHERE cd.car_id = ?;";
```

Comment:
```suggestion
        String query = "SELECT cd.car_id, cd.driver_id, d.name as driver_name, d.license_number "
                + "FROM drivers d "
                + "JOIN cars_drivers cd ON cd.driver_id = d.id "
                + "WHERE cd.car_id = ?";
```
You can remove it everywhere

Reply:
okau!